File: _old/src/KubeOps.Templates/Templates/Operator.FSharp/Entities/V1DemoEntity.fs
namespace GeneratedOperatorProject.Entities

open k8s.Models
open KubeOps.Operator.Entities

type V1DemoEntitySpec() =
    member val Username = "" with get, set

type V1DemoEntityStatus() =
    member val DemoStatus = "" with get, set

[<KubernetesEntity(Group = "demo.kubeops.dev", ApiVersion = "v1", Kind = "DemoEntity")>]
type V1DemoEntity() =
    inherit CustomKubernetesEntity<V1DemoEntitySpec, V1DemoEntityStatus>()

File: _old/src/KubeOps.Templates/Templates/Operator.CSharp/Entities/V1DemoEntity.cs
using k8s.Models;
using KubeOps.Operator.Entities;

namespace GeneratedOperatorProject.Entities;

[KubernetesEntity(Group = "demo.kubeops.dev", ApiVersion = "v1", Kind = "DemoEntity")]
public class V1DemoEntity : CustomKubernetesEntity<V1DemoEntity.V1DemoEntitySpec, V1DemoEntity.V1DemoEntityStatus>
{
    public class V1DemoEntitySpec
    {
        public string Username { get; set; } = string.Empty;
    }

    public class V1DemoEntityStatus
    {
        public string DemoStatus { get; set; } = string.Empty;
    }
}
